cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompilerImpl.java

```diff
@@ -102,10 +102,14 @@ public ExecutableModellingResult preCompileSource(SlangSource source) {
         Validate.notNull(source, "You must supply a source to compile");
 
         //first thing we parse the yaml file into java maps
-        ParsedSlang parsedSlang = yamlParser.parse(source);
+        ParseModellingResult parseModellingResult = yamlParser.parseAndValidate(source);
 
         // Then we transform the parsed Slang source to a Slang model
-        return slangModeller.createModel(parsedSlang);
+        ExecutableModellingResult executableModellingResult =
+                slangModeller.createModel(parseModellingResult.getParsedSlang());
+
+        executableModellingResult.getErrors().addAll(parseModellingResult.getErrors());
```

Comment:
maybe you could move the above line in the slangModeller.createModel method, WDYT?